### PR TITLE
Build pipeline

### PR DESCRIPTION
**Pull request by doomspork**
Ready for review! This PR does a few things for the project:

1. Remove unnecessary Docker and Avro assets. If/when we support avsc file parsing -> schema generation, we can add assets them with the latest tools shipping with Avro.
1. Add Dialyzer. This cause an unused cased I was able to remove.
1. Add Credo and incorporate it's feedback
1. Add TravisCI to run tests, Dialyzer, and Credo.

### CODE REVIEW

File: lib/avro_ex/decode.ex [hunk 1]
```diff
@@ -215,21 +215,19 @@ defmodule AvroEx.Decode do
   end
 
   def variable_integer_decode(<<0::1, n::7, rest::bitstring>>, acc, %Primitive{type: :integer})
-      when is_bitstring(acc) and is_bitstring(acc) do
```
Comment (doomspork) [inline lib/avro_ex/decode.ex]: `when is_bitstring(acc) and is_bitstring(acc)` — identical guards.

File: .formatter.exs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # Used by "mix format"
 [
-  inputs: ["mix.exs", "{config,lib,test}/**/*.{ex,exs}"]
+  inputs: ["mix.exs", "{config,lib,test}/**/*.{ex,exs}"],
+  line_length: 120
```
Comment (doomspork) [inline .formatter.exs]: While we won't enforce using the formatter some people do like to use it, let's head off lines being shortened unnecessarily.

**Comment by doomspork** [comment]
@gerbal you've been contributing a bunch recently, any thoughts on this PR?

File: .credo.exs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        {Credo.Check.Warning.UnusedStringOperation, []},
+        {Credo.Check.Warning.UnusedTupleOperation, []},
+
+        #
+        # Controversial and experimental checks (opt-in, just replace `false` with `[]`)
+        #
+        {Credo.Check.Consistency.MultiAliasImportRequireUse, false},
+        {Credo.Check.Design.DuplicatedCode, false},
+        {Credo.Check.Readability.MultiAlias, false},
+        {Credo.Check.Readability.Specs, false},
```
Comment (gerbal) [inline .credo.exs]: I like having this check enabled, but tastes vary.

Comment (doomspork) [reply .credo.exs]: @gerbal I would agree and I'd also like to enable `@moduledoc`. The _only_ reason I haven't: Credo fails the build and I don't want this PR to go crazy.

Would you be open to following up with a PR to enable this and satisfy the complaints once this is merged?

Comment (doomspork) [reply .credo.exs]: @michaelstalker would you be willing to tackle enabling `@moduledoc` rule and satisfying its complaints?

Comment (michaelstalker) [reply .credo.exs]: I'll try to do that this weekend.

File: lib/avro_ex/encode.ex [hunk 4]
```diff
@@ -16,7 +16,6 @@ defmodule AvroEx.Encode do
     case do_encode(schema, context, data) do
       {:error, :data_does_not_match_schema, _data, _schema} = err -> err
       {:error, _reason, _value} = err -> err
-      {:error, _reason} = err -> err
```
Comment (michaelstalker) [inline lib/avro_ex/encode.ex]: Could this error case ever happen?

File: mix.exs [hunk 5]
```diff
@@ -16,8 +16,7 @@ defmodule AvroEx.Mixfile do
   end
 
   def application do
-    # Specify extra applications you'll use from Erlang/Elixir
-    [extra_applications: [:logger, :ecto]]
+    [extra_applications: [:ecto, :logger]]
```
Comment (michaelstalker) [inline mix.exs]: Do we need `:ecto` in `extra_applications` at all? I thought Elixir loaded all apps in the `deps` list, and that [`extra_applications` was for apps not included in `deps`](https://hexdocs.pm/mix/Mix.Tasks.Compile.App.html#moduledoc).

File: lib/avro_ex/encode.ex (hunk 4 above)
Comment (doomspork) [reply lib/avro_ex/encode.ex]: Dialyzer says no, that's why I removed it.

File: mix.exs (hunk 5 above)
Comment (doomspork) [reply mix.exs]: I think this may be a left over from earlier Elixir versions. Let me confirm and I will remove.

**Comment by doomspork** [comment]
@michaelstalker the existing files weren't included in the release so anyone using the package would have been using them from the repo.

CJ mentioned they had plans to one day support generation of the schema from AVCS on compilation, I think this may have been part of that effort.

**Comment by doomspork** [comment]
Any thoughts on this @beam-community/avro_ex? I want to get this merged in so we can cut a new release for #23

**Comment by CJPoll** [comment]
Your hunch about the test avsc files is correct - I had hoped to make a code generator and they were the first step towards that. They were also useful when testing things manually before writing automated tests, especially before I knew how I wanted to deserialize things. I have no qualms about removing them.

**Pull request merged**